contracts/src/ModuleCRC21.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.6.8;
+
+import "./ModuleCRC20.sol";
+
+contract ModuleCRC21 is ModuleCRC20  {
+
+    event __CronosSendToEthereum(address sender, address recipient, uint256 amount, uint256 bridge_fee);
+    event __CronosCancelSendToEthereum(address sender, uint256 id);
+
+    constructor(string memory denom_, uint8 decimals_) ModuleCRC20(denom_, decimals_) public {
+        decimals = decimals_;
+        denom = denom_;
+    }
+
+    // make unsafe_burn internal
+    function unsafe_burn_v2(address addr, uint amount) internal {
+        // Deduct user's balance without approval
+        require(balanceOf[addr] >= amount, "ds-token-insufficient-balance");
+        balanceOf[addr] = sub(balanceOf[addr], amount);
+        totalSupply = sub(totalSupply, amount);
+        emit Burn(addr, amount);
+    }
+
+    // send to ethereum through gravity bridge
+    function send_to_ethereum_v2(address recipient, uint amount, uint bridge_fee) external {
```

Comment:
Without migration, it is possible to create a wrapper over the CRC20 to be able to call `__CronosSendToEthereum` and have a separate supply from the original CRC20

It would be cleaner to remove `send_to_ethereum` from the CRC20.sol and remove the _v2 here however this might change the bytecode of the contract (even though nobody is using this method now)

Any opinion @tomtau @yihuang

Reply:
Agree with the standard extension, and deprecate the `send_to_ethereum` from the original one.
But existing contracts still need to migrate code to support the gravity bridge, right?

Reply:
another idea is to have a wrapper over exsting CRC20 contract to be able to add the ability to bridge. 
For example having  `ETH_VVS` will be a wrapper of `VVS.sol` and expose `send_to_ethereum` and `wrapping/unwrapping` functions
The 1-1 supply control will be guarantee by the smart contract

Reply:
I'd suggest documenting these architectural decisions: https://github.com/crypto-org-chain/cronos/blob/main/docs/architecture/adr-template.md + getting some feedback from dApp developers on it

Reply:
> another idea is to have a wrapper over exsting CRC20 contract to be able to add the ability to bridge. For example having `ETH_VVS` will be a wrapper of `VVS.sol` and expose `send_to_ethereum` and `wrapping/unwrapping` functions The 1-1 supply control will be guarantee by the smart contract

it's still a contract migration, the client need to use a different contract address, but internally it reuse the old contract to avoid state migration, right?

Reply:
```the client need to use a different contract address, but internally it reuse the old contract to avoid state migration```

Why does it need to use old contract? Mapping between gravitydenom <-> contract address will be done on the wrapped contract.
The drawback is that to be able to send to ethereum, one needs to wrap its token first (or unwrap for the opposite direction) but that can be integrated seeminglessly by the Dapp

Reply:
I see, you mean a 1-to-1 token wrapping contract, I guess for erc20, user will need to do a) approve, b) swap, c) send_to_ethereum, right?